src/patterns.md

```diff
@@ -694,7 +716,11 @@ match int_reference {
 > &nbsp;&nbsp; `[` _SlicePatternItems_<sup>?</sup> `]`
 >
 > _SlicePatternItems_ :\
-> &nbsp;&nbsp; [_Pattern_] \(`,` [_Pattern_])<sup>\*</sup> `,`<sup>?</sup>
+> &nbsp;&nbsp; _InSlicePattern_ \(`,` _InSlicePattern_)<sup>\*</sup> `,`<sup>?</sup>
+>
+> _InSlicePattern_ :\
+> &nbsp;&nbsp; &nbsp;&nbsp; [_PatternWithoutRange_]\
+> &nbsp;&nbsp; | [_ClosedRangePattern_]
```

Comment:
I believe the restriction is done after parsing, so this shouldn't be expressed as part of the grammar.  That is, `[0..]` is syntactically valid as a pattern, but is then later rejected.  I think it would be fine just to list that as a restriction below in prose.

(and, fwiw, _ClosedRangePattern_ isn't defined)

Reply:
Fair enough then.
Whoops.
Alright, let's see...

Reply:
Also a good point because: I forgot that it was only the version _without parentheses_ that is currently "up in the air".